OpenRA.Mods.Common/Traits/Attack/AttackDetonates.cs

```diff
@@ -77,11 +70,23 @@ public void ResolveOrder(Actor self, Order order)
 					self.CancelActivity();
 
 				self.SetTargetLine(target, Color.Red);
-				self.QueueActivity(new MoveAdjacentTo(self, target));
-				self.QueueActivity(new CallFunc(() => Explode(self)));
+
+				if (self.Info.HasTraitInfo<MobileInfo>())
+					self.QueueActivity(new MoveAdjacentTo(self, target));
+				else
+				{
+					var aircraft = self.TraitOrDefault<Aircraft>();
+					if (aircraft != null)
```

Comment:
No, it can't be done. It would crash later on. https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Activities/Move/MoveAdjacentTo.cs#L56 queries `Mobile` and then `MobileInfo` for the movement class.